src/libs/ReportActionsUtils.ts

```diff
@@ -39,202 +46,142 @@ Onyx.connect({
-                return (first.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW ? 1 : -1) * invertedMultiplier;
-            }
-
-            // Then fallback on reportActionID as the final sorting criteria. It is a random number,
-            // but using this will ensure that the order of reportActions with the same created time and action type
-            // will be consistent across all users and devices
-            return (first.reportActionID < second.reportActionID ? -1 : 1) * invertedMultiplier;
-        })
-        .value();
+
+    return reportActions.filter(Boolean).sort((first, second) => {
+        // First sort by timestamp
+        if (first.created !== second.created) {
+            return (first.created < second.created ? -1 : 1) * invertedMultiplier;
+        }
+
+        // Then by action type, ensuring that `CREATED` actions always come first if they have the same timestamp as another action type
+        if ((first.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED || second.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED) && first.actionName !== second.actionName) {
+            return (first.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED ? -1 : 1) * invertedMultiplier;
+        }
+        // Ensure that `REPORTPREVIEW` actions always come after if they have the same timestamp as another action type
+        if ((first.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW || second.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW) && first.actionName !== second.actionName) {
+            return (first.actionName === CONST.REPORT.ACTIONS.TYPE.REPORTPREVIEW ? 1 : -1) * invertedMultiplier;
+        }
+
+        // Then fallback on reportActionID as the final sorting criteria. It is a random number,
+        // but using this will ensure that the order of reportActions with the same created time and action type
+        // will be consistent across all users and devices
+        return (first.reportActionID < second.reportActionID ? -1 : 1) * invertedMultiplier;
+    });
 }
 
 /**
  * Finds most recent IOU request action ID.
- *
- * @param {Array} reportActions
- * @returns {String}
  */
-function getMostRecentIOURequestActionID(reportActions) {
-    const iouRequestTypes = [CONST.IOU.REPORT_ACTION_TYPE.CREATE, CONST.IOU.REPORT_ACTION_TYPE.SPLIT];
-    const iouRequestActions = _.filter(reportActions, (action) => iouRequestTypes.includes(lodashGet(action, 'originalMessage.type')));
+function getMostRecentIOURequestActionID(reportActions: ReportAction[] | null): string | null {
+    const iouRequestTypes: Array<ValueOf<typeof CONST.IOU.REPORT_ACTION_TYPE>> = [CONST.IOU.REPORT_ACTION_TYPE.CREATE, CONST.IOU.REPORT_ACTION_TYPE.SPLIT];
+    const iouRequestActions = reportActions?.filter((action) => action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && iouRequestTypes.includes(action.originalMessage.type)) ?? [];
```

Comment:
Seems like this line after fresh sing in evaluates to an object `{}` which then leads to crash https://github.com/Expensify/App/issues/29965